Mathlib/Algebra/Algebra/NonUnitalSubalgebra.lean

```diff
@@ -976,7 +976,7 @@ noncomputable def iSupLift [Nonempty ι] (K : ι → NonUnitalSubalgebra R A) (d
               simp only
               rw [hf i k hik, hf j k hjk]
               rfl)
-            (↑(iSup K)) (by rw [coe_iSup_of_directed dir])
+            _ (by rw [coe_iSup_of_directed dir])
```

Comment:
Out of curiosity, what's happening here? Most of the changes in this PR are making things more explicit - which is what I expected given the change in lean4#6024, but here it's becoming less explicit!

Reply:
It turns out this one was failing because the coercion created a metavariable that stopped the propagation of any information. The solutions were to either (1) figure out what the actual expected type was supposed to be (often this was hard to write down for various reasons!) or (2) erase the expression since it was determined by the other arguments anyway.

I'm really hoping that we'll have a more principled way to propagate type information across coercions. This fix is a necessary regression for now.

Reply:
Interesting! Thanks for the explanation